File: day-02/day_02_test.py
import os
from pathlib import Path
import pytest
from day_02 import checksum, common_letters


def test_part_one_example():
    test_data = ["abcdef", "bababc", "abbcde", "abcccd", "aabcdd", "abcdee", "ababab"]
    assert checksum("\n".join(test_data)) == 12


def test_part_one_answer():
    data = Path("day-02/input.txt").read_text()
    assert checksum(data) == 6972


@pytest.mark.skip(reason="todo")
def test_part_two_example():
    data = "\n".join(["abcde", "fghij", "klmno", "pqrst", "fguij", "axcye", "wvxyz"])
    assert common_letters(data) == "fgij"

File: src/day-02.test.ts
import { checksum, commonLetters, input } from './day-02'

test('part one example', () => {
  expect(
    checksum(
      [
        'abcdef',
        'bababc',
        'abbcde',
        'abcccd',
        'aabcdd',
        'abcdee',
        'ababab',
      ].join('\n')
    )
  ).toEqual(12)
})

test('part one answer', () => {
  expect(checksum(input)).toEqual(6972)
})

test('part two example', () => {
  let exampleInput = `abcde
fghij
klmno
pqrst
fguij
axcye
wvxyz`
  expect(commonLetters(exampleInput)).toEqual('fgij')
})

test('part two solution', () => {
  expect(commonLetters(input)).toEqual('aixwcbzrmdvpsjfgllthdyoqe')
})
